### PR TITLE
feat: publish assessment from create assessment page

### PR DESCRIPTION
**Pull request by joyce-shi**
## Notion ticket link

[Publish Assessment in Create page](https://www.notion.so/uwblueprintexecs/Publish-Assessment-in-Create-page-97c1e33243224b9ba9c121b2c714e34f?pvs=4)

## Implementation description
* added handler for publish assessment

## Steps to test
1. create a new test
2. click publish
3. verify that this creates a published test

## What should reviewers focus on?
* functionality works as expected
* error handling works as expected
* existing publish assessment from display page works as expected

### CODE REVIEW

**Comment by joyce-shi** [comment]
cc @cindy-531

File: frontend/src/components/pages/admin/CreateAssessmentPage.tsx [hunk 1]
```diff
@@ -49,17 +49,13 @@ const CreateAssessmentPage = (): React.ReactElement => {
     }
   }, [questions, errorMessage]);
 
-  const onSave: SubmitHandler<TestRequest> = async (data) => {
+  const onCreateTest = async (test: TestRequest) => {
```
Comment (carissa-tang) [inline frontend/src/components/pages/admin/CreateAssessmentPage.tsx]: should we just have `onCreateTest` take in a status field? then we don't need `onSave` and `onPublish`

File: frontend/src/components/assessments/assessment-creation/CreateAssessmentHeader.tsx [hunk 2]
```diff
@@ -16,56 +16,68 @@ import {
 } from "../../../assets/icons";
 import { getReadableDate } from "../../../utils/GeneralUtils";
 import BackButton from "../../common/BackButton";
+import PublishModal from "../assessment-status/EditStatusModals/PublishModal";
 
 interface CreateAssessementHeaderProps {
   name: string;
-  onSave: (event: React.MouseEvent<HTMLButtonElement>) => void;
+  onSave: () => void;
+  onPublish: () => void;
```
Comment (carissa-tang) [inline frontend/src/components/assessments/assessment-creation/CreateAssessmentHeader.tsx]:
```suggestion
  onSave: () => Promise<void>;
  onPublish: () => Promise<void>;
```
i think

File: frontend/src/components/assessments/assessment-status/EditStatusModals/PublishModal.tsx [hunk 3]
```diff
@@ -1,43 +1,20 @@
...
+  publishAssessment: () => void;
 }
 
 const PublishModal = ({
   isOpen,
   onClose,
-  assessmentId,
+  publishAssessment,
 }: PublishModalProps): React.ReactElement => {
-  const [publishAssessment, { error }] = useMutation<{
```
Comment (carissa-tang) [inline frontend/src/components/assessments/assessment-status/EditStatusModals/PublishModal.tsx:20]: is there a reason we moved this logic to the button? can we also keep the other files consistent (move delete modal logic to delete button)

**Review by carissa-tang — COMMENTED** [review]
should the publish button validate the form before showing the modal? (same behaviour as save)

File: frontend/src/components/pages/admin/CreateAssessmentPage.tsx (hunk 1 above)
Comment (joyce-shi) [reply frontend/src/components/pages/admin/CreateAssessmentPage.tsx]: i think it doesn't work b/c the `SubmitHandler` can't take in multiple parameters? cuz you have to pass to `handleSubmit` in the provided function signature format

File: frontend/src/components/assessments/assessment-creation/CreateAssessmentHeader.tsx [hunk 4]
```diff
@@ -9,63 +10,92 @@ import {
...
+  const [showPublishModal, setShowPublishModal] = React.useState(false);
+  const confirmPublish = () => {
+    if (validateForm()) {
+      setShowPublishModal(true);
+    }
+  };
+
+  const handleSave = handleSubmit(onSave, onError);
+  const handlePublish = handleSubmit(confirmPublish, onError);
+  const onConfirmPublish = handleSubmit(onPublish, onError);
```
Comment (carissa-tang) [inline frontend/src/components/assessments/assessment-creation/CreateAssessmentHeader.tsx]:
```suggestion
  const handlePublish = handleSubmit(onPublish, onError);
  const onConfirmPublish = handleSubmit(onConfirmPublish, onError);
```
i think that makes more sense to me? (renaming)

**Review by carissa-tang — APPROVED** [review]

**Pull request merged**